core/iwasm/libraries/lib-pthread/lib_pthread_wrapper.c

```diff
@@ -174,6 +206,11 @@ lib_pthread_init()
         os_mutex_destroy(&thread_global_lock);
         return false;
     }
+    if (!(sem_info_map = bh_hash_map_create(32, true, (HashFunc)sem_name_hash,
+                                            thread_handle_equal, NULL,
+                                            thread_info_destroy))) {
```

Comment:
Should handle case semaphore in thread_info_destroy, or the resource for semaphore won't be destroyed.